docs/map-making/how-to-upgrade-legacy-map-features.md

```diff
@@ -41,43 +32,20 @@ Within every game (XML) file, inside the "games" folder inside the main folder o
 - `="Battleships repair at beginning of round"` >> `="Units Repair Hits Start Turn"`
 - `="Naval Bombard Casualties Return Fire Restricted"` >> `="Naval Bombard Casualties Return Fire"`
 - `="SBR Affects Unit Production"` >> `="Damage From Bombing Done To Units Instead Of Territories"`
-
-### Changes of game (XML) file special code for the above items:
-
-_(these are cases of advanced coding, so you probably don't have these within the game file)_
-
-Within every game file (XML), inside the "games" folder inside the main folder of the map, change:
-
 - `="conditionType" count="XOR"` >> `="conditionType" count="1"`
 - `="conditionType" count="-reset-XOR"` >> `="conditionType" count="-reset-1"`
-
-### Changes of already deprecated game (XML) file items:
-
-_(these items were already deprecated, so you shouldn't have had these within the game file already)_
-
-Within every game (XML) file, inside the "games" folder inside the main folder of the map, change:
-
-- `="isTwoHit" value="true"` >> `="hitPoints" value="2"`
-- `="isTwoHit" value="false"` >> `="hitPoints" value="1"`
-- `="occupiedTerrOf"` >> `="originalOwner"`
-- `="victoryCity" value="true"` >> `="victoryCity" value="1"`
-- `="victoryCity" value="false"` >> `="victoryCity" value="0"`
-
-### Changes of game (XML) file special code for the above items:
-
-_(these are cases of advanced coding, so you probably don't have these within the game file)_
-
-Within every game file (XML), inside the "games" folder inside the main folder of the map, change:
-
 - `="isTwoHit" count="true"` >> `="hitPoints" count="2"`
+- `="isTwoHit" value="false"` >> `="hitPoints" value="1"`
```

Comment:
This item, or any item, should better not stay in the middle of the "reset" and "non-reset" version of the code for turning the "isTwoHit" option value to true (the "reset" is a redundant coding, since the option accepts only one value, but it cannot be exclude a game might have it).

Reply:
The biggest consumer of this document so far, probably has been me. Considering I used it to port legacy game features to non-legacy in the game engine. Though, I found the document to not be useful and had to go back to the 10 page conversation and to great pains parse out what had been changed. Really I want a simple list of:
`a -> b`, and beyond that an absolute minimum of explanatory text and other corner cases that are not important.

I state the above also after the observation that someone explicitly asking to upgrade maps was directed to this document and it was not useful for them. Hence, the motivation to keep it simple and a straight forward change log of old to new. It may even turn out to be the case that this becomes just a reference document as well if the engine can automatically upgrade maps for map-makers.

So with that said, could you clarify why the ordering is important and/or how the before and after instruction needs improvement? Can you illustrate an example where this would steer someone wrong?

Reply:
The order is not needed, as you can make the changes in any order, but I think what is similar the most is better to stay close. This principle appears to be generally followed in the document, as well. For the changes at hand, I suggest the following order, if we have a single section:
- `="isTwoHit" value="true"` >> `="hitPoints" value="2"`
- `="isTwoHit" value="false"` >> `="hitPoints" value="1"`
- `="isTwoHit" count="true"` >> `="hitPoints" count="2"`
- `="isTwoHit" count="-reset-true"` >> `="hitPoints" count="-reset-2"`
- `="isTwoHit" count="false"` >> `="hitPoints" count="1"`
- `="isTwoHit" count="-reset-false"` >> `="hitPoints" count="-reset-1"`

And so on.
As I said, we could mix up the above in whatever order and intermix any of them with other code changes, but I think that would be unpolished. It is simply better for what changes the same option to be next to each other.

Reply:
Pointing out that I advise against what I'm saying now (to keep if friendly for persons not able to handle advanced search and replace), the above 6 lines of codes could have been exchanged for the following two (assuming "\*" means "zero or more of any"):
- `="isTwoHit"*="*true"` >> `="hitPoints"*="*2"`
- `="isTwoHit"*="*false"` >> `="hitPoints"*="*1"`